src/client/opamSwitchCommand.ml

```diff
@@ -477,38 +477,51 @@ let import_t ?ask importfile t =
   end;
   t
 
-let read_overlays (read: package -> OpamFile.OPAM.t option) packages =
+let read_overlays ~repos_roots (read: package -> OpamFile.OPAM.t option) packages =
   OpamPackage.Set.fold (fun nv acc ->
       match read nv with
       | Some opam ->
-        if OpamFile.OPAM.extra_files opam <> None then
-          (OpamConsole.warning
-             "Metadata of package %s uses a files%s subdirectory, it may not be \
-              re-imported correctly (skipping definition)"
-             (OpamPackage.to_string nv) Filename.dir_sep;
-           acc)
-        else OpamPackage.Name.Map.add nv.name opam acc
+        let opam' = match OpamFile.OPAM.get_extra_files ~repos_roots opam with
+          | [] -> opam
+          | files ->
+            (* read file and embed them into opam *)
+            List.fold_left (fun opam (file, _rel_file, hash) ->
+                if OpamFilename.exists file &&
+                   OpamHash.check_file (OpamFilename.to_string file) hash then
+                  let value = OpamFilename.read file in
+                  let value' = B64.encode_string value in
+                  let name = "x-" ^ OpamHash.contents hash in
```

Comment:
`x-extra-file-HASH` is it now, I'd prefer not to use a single field containing pairs, since that means I'd need to write more code. I can't use `x-filename` since filename is not necessarily an identifier.